internal/tiltfile/docker_compose.go

```diff
@@ -93,11 +93,12 @@ func (s *tiltfileState) dcResource(thread *starlark.Thread, fn *starlark.Builtin
 		return nil, fmt.Errorf("dc_resource: `name` must not be empty")
 	}
 
-	var imageRefAsStr string
+	var imageRefAsStr *string
 	switch imageVal := imageVal.(type) {
 	case nil: // optional arg, this is fine
```

Comment:
...wait, does this `case nil` ever trip? seems like the least change here would be to, in the nil case, set `imageRefAsStr = ""` and check below if it's the empty string, no?